core/src/main/java/org/apache/accumulo/core/fate/FateExecutor.java

```diff
@@ -412,7 +418,9 @@ public void run() {
         }
       } finally {
         log.trace("A TransactionRunner is exiting...");
-        Preconditions.checkState(runningTxRunners.remove(this));
+        synchronized (runningTxRunners) {
+          Preconditions.checkState(runningTxRunners.remove(this));
+        }
```

Comment:
no need to sync here since `this.runningTxRunners = Collections.synchronizedSet(new HashSet<>());`

Reply:
But we are synchronizing on it elsewhere...